File: ej01_segundos_a_hms/typescript/solucion.ts
type Hms = {
    horas: number
    minutos: number
    segundos_finales: number
}

export const segundosAHms = (segundos: number): Hms => {

    const horas: number = Math.floor(segundos / 3600);
    const resto: number = segundos % 3600;

    const minutos: number = Math.floor(resto / 60);
    const segundos_finales: number = resto % 60;

    return { horas, minutos, segundos_finales }
}

File: ej01_segundos_a_hms/python/solucion.py
def segundos_a_hms(segundos: int) -> tuple[int, int, int]:

    horas = segundos // 3600
    resto = segundos % 3600

    minutos = resto // 60
    segundos_finales = resto % 60

    return horas, minutos, segundos_finales
